Source/Route.swift

```diff
@@ -16,35 +16,31 @@
 
 import UIKit
 
-public class Route {
+class Route {
     
-    static let routeParameterPattern = ":[a-zA-Z0-9-_]+"
-    static let urlParameterPattern = "([^/]+)"
+    enum Pattern: String {
+        case RouteParam = ":[a-zA-Z0-9-_]+"
+        case UrlParam = "([^/]+)"
+    }
     
-    let routeParameter: NSRegularExpression = try! NSRegularExpression(pattern: routeParameterPattern, options: .CaseInsensitive)
-    let urlParameter: NSRegularExpression = try! NSRegularExpression(pattern: urlParameterPattern, options: .CaseInsensitive)
+    let routeParameter: NSRegularExpression = try! NSRegularExpression(pattern: Pattern.RouteParam.rawValue, options: .CaseInsensitive)
+    let urlParameter: NSRegularExpression = try! NSRegularExpression(pattern: Pattern.UrlParam.rawValue, options: .CaseInsensitive)
```

Comment:
We can do a few things here, we can remove the force try to avoid crashes but we can also create an extension on NSRegularExpression to accept our enum so we do not need to unwrap it using rawValue here.

```
extension NSRegularExpression {
  
  convenience init(pattern: Route.Pattern, options: NSRegularExpressionOptions) throws {
    try self.init(pattern: pattern.rawValue, options: options)
  }
  
}
```

Then we can neaten up the code above using our new extension.

```
let routeParameter = try! NSRegularExpression(pattern: .RouteParam, options: .CaseInsensitive)
let urlParameter   = try! NSRegularExpression(pattern: .UrlParam, options: .CaseInsensitive)
```

We could look at removing the force unwrap but it is not necessary here since the input is not changing